### PR TITLE
No reset radio button

### PR DESCRIPTION
**Pull request by TuckerSherman**
Small fix preventing users from selecting reset and no reset at the same time from iOS or Android settings

### CODE REVIEW

File: Appium/Model/AppiumAndroidSettingsModel.m [hunk 1]
```diff
@@ -86,89 +89,197 @@ -(NSArray*) allPlatformVersions { return [NSArray arrayWithObjects:
...
+	[[NSUserDefaults standardUserDefaults] setValue:deviceReadyTimeout forKey:APPIUM_PLIST_ANDROID_DEVICE_READY_TIMEOUT];
+}
 
--(BOOL) fullReset { return [DEFAULTS boolForKey:APPIUM_PLIST_ANDROID_FULL_RESET]; }
--(void) setFullReset:(BOOL)fullReset { [DEFAULTS setBool:fullReset forKey:APPIUM_PLIST_ANDROID_FULL_RESET]; }
+-(BOOL) fullReset {
+	return [DEFAULTS boolForKey:APPIUM_PLIST_ANDROID_FULL_RESET];
+}
+-(void) setFullReset:(BOOL)fullReset {
+	if((self.noReset==YES)&&(fullReset==YES)){
```
Comment (edgy360) [inline Appium/Model/AppiumAndroidSettingsModel.m]: There is no need for the `==YES` here. Please could you change it to:
```
if (self.noReset && fullReset) {
```

File: Appium/Model/AppiumAndroidSettingsModel.m [hunk 2]
```diff
@@ -86,89 +89,197 @@ -(NSArray*) allPlatformVersions { return [NSArray arrayWithObjects:
...
+		[self setNoReset:NO];
+	}
+	[DEFAULTS setBool:fullReset forKey:APPIUM_PLIST_ANDROID_FULL_RESET];
+}
+
+-(BOOL) noReset {
+	return [DEFAULTS boolForKey:APPIUM_PLIST_ANDROID_NO_RESET];
+}
+-(void) setNoReset:(BOOL)noReset {
+	if((self.fullReset==YES)&&(noReset==YES)){
```
Comment (edgy360) [inline Appium/Model/AppiumAndroidSettingsModel.m]: As per the same comment for line 177 (`==YES`).

File: Appium/Model/AppiumiOSSettingsModel.m [hunk 3]
```diff
@@ -59,120 +58,261 @@ -(NSArray*) allPlatformVersions {
...
+	[DEFAULTS setValue:locale forKey:APPIUM_PLIST_IOS_LOCALE];
+}
 
--(BOOL) noReset { return [DEFAULTS boolForKey:APPIUM_PLIST_IOS_NO_RESET]; }
--(void) setNoReset:(BOOL)noReset { [DEFAULTS setBool:noReset forKey:APPIUM_PLIST_IOS_NO_RESET]; }
+-(BOOL) noReset {
+	return [DEFAULTS boolForKey:APPIUM_PLIST_IOS_NO_RESET];
+}
+-(void) setNoReset:(BOOL)noReset {
+	if((noReset==YES) && (self.fullReset==YES)){
```
Comment (edgy360) [inline Appium/Model/AppiumiOSSettingsModel.m]: As per the same comment for line 177 (`==YES`).

File: Appium/Model/AppiumiOSSettingsModel.m [hunk 4]
```diff
@@ -59,120 +58,261 @@ -(NSArray*) allPlatformVersions {
...
+		[self setFullReset:NO];
+	}
+	[DEFAULTS setBool:noReset forKey:APPIUM_PLIST_IOS_NO_RESET];
+}
+-(BOOL) fullReset {
+	return [DEFAULTS boolForKey:APPIUM_PLIST_IOS_FULL_RESET];
+}
+-(void) setFullReset:(BOOL)fullReset {
+	if((self.noReset==YES) && (fullReset==YES)){
```
Comment (edgy360) [inline Appium/Model/AppiumiOSSettingsModel.m]: As per the same comment for line 177 (`==YES`).

**Comment by edgy360** [comment]
Thanks for submitting this pull request. This is a really useful fix.

I added a few line comments about the `==YES`, which is only a minor fix.

The only other part to mention is around the layout of `{ }`'s. We are trying to go with the approach of the brackets on a new line. Please could you update the pull request to reflect this too?
```
- (void)someMethod
{
	if (someCondition)
	{
	
	}
}
```

**Comment by TuckerSherman** [comment]
Can do! thanks!

**Comment by edgy360** [comment]
Thanks! Sorry I didn't merge, I thought that GitHub would email me when new commits were pushed.

**Pull request merged**